alpm-utils/src/alpm.rs

```diff
@@ -1,22 +1,71 @@
-use crate::depends::satisfies_ver;
+//! Extension methods for the [`Alpm`] type.
 
-use alpm::{Alpm, Result, Package, Depend};
+use crate::depends::satisfies_ver;
+use crate::DbListExt;
+use alpm::{Alpm, Depend, Package, PackageReason, SigLevel};
+use std::path::Path;
 
+/// Extension methods to the [`Alpm`] type.
 pub trait AlpmExt {
-    fn find_local_satisfier<S: Into<String>>(&self, pkg: S) -> Result<Option<Package>>;
+    /// Try to find a [`Package`] that satisfies a given dependency.
+    fn find_local_satisfier<S>(&self, pkg: S) -> Option<Package<'_>>
+    where
+        S: Into<String>;
 }
 
 impl AlpmExt for Alpm {
-    fn find_local_satisfier<S: Into<String>>(&self, pkg: S) -> Result<Option<Package>> {
+    fn find_local_satisfier<S>(&self, pkg: S) -> Option<Package<'_>>
+    where
+        S: Into<String>,
+    {
         let localdb = self.localdb();
         let pkg = pkg.into();
 
-        if let Ok(alpm_pkg) = localdb.pkg(&pkg) {
-            if satisfies_ver(&Depend::new(&pkg), alpm_pkg.version()) {
-                return Ok(Some(alpm_pkg));
+        match localdb.pkg(pkg.as_str()) {
+            Ok(alpm_pkg) if satisfies_ver(Depend::new(pkg.as_str()), alpm_pkg.version()) => {
+                Some(alpm_pkg)
             }
+            _ => localdb.pkgs().find_satisfier(pkg),
         }
+    }
+}
+
+/// All orphaned packages.
+///
+/// An orphan is a package that was installed as a dependency, but whose parent
+/// package is no longer installed.
+pub fn orphans(alpm: &Alpm) -> impl Iterator<Item = Package<'_>> {
```

Comment:
This is too opinionated to belong here IMO. This won't watch dependency cycles, an issue pacman also suffers from.